lib/proxy.js

```diff
@@ -192,50 +200,85 @@ function ReverseProxy(opts){
   }
 }
 
+ReverseProxy.prototype.addResolver = function (resolver) {
+  if(this.opts.cluster && cluster.isMaster) return this;
+
+  if(!_.isArray(resolver)){
+    resolver = [resolver];
+  }
+
+  var _this = this;
+  resolver.forEach(function (resolveObj) {
+    if(!_.isFunction(resolveObj)){
+      throw new Error("Resolver must be an invokable function.");
+    }
+
+    if(!resolveObj.hasOwnProperty('priority')){
+      resolveObj.priority = 0;
+    }
+
+    _this.resolvers.push(resolveObj);
+  });
+
+  _.sortBy(_this.resolvers, function (r) {
+    return -r.priority;
+  });
+};
+
+ReverseProxy.prototype.removeResolver = function (resolver) {
+  if(this.opts.cluster && cluster.isMaster) return this;
+  //since unique resolvers are not checked for performance,
+  //just remove every existence.
+  this.resolvers = this.resolvers.filter(function (resolverFn) {
+    return resolverFn !== resolver;
+  });
+};
+
+ReverseProxy.buildTarget = function (target, opts) {
+  opts = opts || {};
+  target = prepareUrl(target);
+  target.sslRedirect =  !opts.ssl || opts.ssl.redirect !== false;
+  target.useTargetHostHeader = opts.useTargetHostHeader === true;
+  return target;
+};
+
 /**
-  Register a new route.
+ Register a new route.
 
-  @src {String|URL} A string or a url parsed by node url module.
-  Note that port is ignored, since the proxy just listens to one port.
+ @src {String|URL} A string or a url parsed by node url module.
+ Note that port is ignored, since the proxy just listens to one port.
 
-  @target {String|URL} A string or a url parsed by node url module.
-  @opts {Object} Route options.
-*/
+ @target {String|URL} A string or a url parsed by node url module.
+ @opts {Object} Route options.
+ */
 ReverseProxy.prototype.register = function(src, target, opts){
   if(this.opts.cluster && cluster.isMaster) return this;
```

Comment:
@logbon72 ok. I understand what you mean. I do not like magic functions either that do a lot of things. However in this case I think it makes sense, because "register" is supposed to be a generic method, but when introducing the concept of resolvers, register is just a way to add a predefined resolver, while addResolver becomes the new generic method. I think I can merge as is though and then change the API after a bit more of thought.